nmea_parser.c

```diff
@@ -128,6 +131,46 @@ void parse_NMEA_command(char* message){
 						strncpy(val,ptr,strlen(ptr));
 						polar.w=atof(val);
 						setPolar(polar.a, polar.b, polar.c, polar.w);
+						debug_print("Get Polar POL: %f, %f, %f, %f\n",
+                                                polar.a,polar.b,polar.c,polar.w);
+					}
+				break;
+
+				case 'I':
+					//Set ideal Polar
+					if (*(ptr+1) == 'P' &&  *(ptr+2) == 'O') {
+						ptr = strtok(NULL, delimiter);
+						val = (char *) malloc(strlen(ptr));
```

Comment:
When this memory is going to be released? This looks like a memory leak.

Reply:
good question. Please ask the original author.
I should replace the malloc()'s with some static buffer.
The code is full of it.